Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,315 @@ public static class BalanceStats
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (IsStrArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            CultureInfo.CurrentCulture = GetCustomInvariantCulture();
+
+            decimal max = decimal.MinValue;
+            string maxResult = "";
+            string peopleWithSameHighBalance = "";
+            bool moreThanOne = false;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] auxBalances = peopleAndBalances[i].Split(',');
+                decimal[] balances = GetBalances(auxBalances[1..]);
+                decimal currentMax = MaxBalance(balances);
+                if (max < currentMax )
+                {
+                    max = currentMax;
+                    peopleWithSameHighBalance = auxBalances[0];
+                    moreThanOne = false;
+                    maxResult = $" had the most money ever. {max:C0}.";
+                }
+               else if (max.Equals(currentMax))
+                {
+                    peopleWithSameHighBalance += $", {auxBalances[0]}";
+                    moreThanOne = true;
+                }
+            }
+
+            if (moreThanOne)
+            {
+                peopleWithSameHighBalance = ReplaceLastCommaWithAnd(peopleWithSameHighBalance);
+            }
+
+            return peopleWithSameHighBalance + maxResult;
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsStrArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            CultureInfo.CurrentCulture = GetCustomInvariantCulture();
+
+            decimal min = decimal.MaxValue;
+            string minResult = "";
+            string peopleWithSameLowBalance = "";
+            bool moreThanOne = false;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] auxBalances = peopleAndBalances[i].Split(',');
+                decimal[] balances = GetBalances(auxBalances[1..]);
+
+                if (balances.Length <= 1)
+                {
+                    return "N/A.";
+                }
+         
+                decimal currentMin = MinBalance(GetDiff(balances));
+                if (min > currentMin)
+                {
+                    min = currentMin;
+                    peopleWithSameLowBalance = auxBalances[0];
+                    moreThanOne = false;
+                    minResult = $" lost the most money. {min:C0}.";
+                }
+                else if (min.Equals(currentMin))
+                {
+                    peopleWithSameLowBalance += $", {auxBalances[0]}";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5069037)